dali/pipeline/executor/executor.cc

```diff
@@ -524,6 +530,12 @@ void Executor<WorkspacePolicy, QueuePolicy>::RunHelper(OpNode &op_node, Workspac
       in.SetLayout({});
     }
   }
+
+  // If it is the end of an epoch, create a checkpoint
+  if (checkpointing_ && (iteration_id + 1) % checkpointing_epoch_size_ == 0) {
+    auto &cpt = GetCurrentIterationData(iteration_id + 1).checkpoint.GetOpCheckpoint(op_node.id);
```

Comment:
Hmm, thinking about it, I would put a comment explaining the fact, that if we write into the next iteration id, consume its outputs and than call the "GetCurrentCheckpoint" we will get the data from here (basically explaining that +1). It might make things easier in the future.